auth.py

```diff
@@ -0,0 +1,49 @@
+import calendar, logging, json
+from datetime import datetime, timedelta
+from jwcrypto import jwt, jwk, jws
+
+
+logger = logging.getLogger(__file__)
+
+def load_token_key(path_token_key):
+    try:
+        with open(path_token_key, 'r') as f:
+            token_key = jwk.JWK.from_json(f.read())
+            return token_key
+    except FileNotFoundError as err:
+        logger.error(f'FileNotFoundError: no credentials.txt in {path_token_key}: {err}')
+        raise FileNotFoundError(f'FileNotFoundError: no credentials.txt in {path_token_key}: {err}')
+    except jwk.InvalidJWKValue as err:
+        logger.error(f'InvalidJWKValue: data in {path_token_key} is not JWK object')     
+        raise TypeError(f'InvalidJWKValue: data in {path_token_key} is not JWK object')   
+    
+class Token_gen:
```

Comment:
You probably want to rename this class to match python naming conventions (`TokenGen` or better `TokenGenerator`)

Reply:
Done!